### PR TITLE
Added OpenWhisk addon for minishift

### PR DESCRIPTION
**Pull request by VeerMuchandi**
Please review and let me know if any issues.

### CODE REVIEW

**Comment by praveenkumar** [comment]
@VeerMuchandi At first look, you need to rebase your branch and also squash the commit into a single commit.

File: add-ons/openwhisk/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+## Prerequisites
+
+Set up minishift with the following configuration and start
+
+```
+ minishift profile set openwhisk-test
+ minishift config set memory 8GB
+ minishift config set cpus 3
+ minishift config set image-caching true
```
Comment (praveenkumar) [inline add-ons/openwhisk/README.md]: This is now by default `true` for every profile.

File: add-ons/openwhisk/openwhisk.addon [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Name: openwhisk 
+# Description: Installs OpenWhisk FaaS 
+# Required-Vars: OPENSHIFT_USERNAME
+# OpenShift-Version: >=3.9.0       
+# Var-Defaults: OPENSHIFT_USERNAME=developer,CONSOLE_URL=https://#{ip}:8443,PROJECT_NAME=openwhisk
+# Url: Template from https://git.io/openwhisk-template 
+###################################################################################################
+
+# login
+oc login -u admin -p admin
```
Comment (praveenkumar) [inline add-ons/openwhisk/openwhisk.addon]: @VeerMuchandi if you don't put login info then by default all the `oc` command in addon will run as `system:admin` and that will do needful I think.

**Comment by VeerMuchandi** [comment]
@praveenkumar do you mean that the below line is default?

`minishift config set image-caching true`

If I don't login as admin, it tries to deploy as `developer`

**Comment by praveenkumar** [comment]
> `minishift config set image-caching true`

@VeerMuchandi yes this is default.

> If I don't login as admin, it tries to deploy as `developer`

That's odd, will check it once.

**Comment by VeerMuchandi** [comment]
@praveenkumar I have only one commit after the last pull. I will close this PR and open a new one.

**Pull request closed without merging**